crates/blockchain/blockchain.rs

```diff
@@ -138,23 +138,43 @@ impl Blockchain {
 
     pub async fn add_block(&self, block: &Block) -> Result<(), ChainError> {
         let since = Instant::now();
-        // Easiest way to operate on the result of `execute_block` without
-        // having to add too much control flow or return early
-        // Async doesn't play well with `.and_then`
-        let inner = || async {
-            let res = self.execute_block(block).await?;
-            self.store_block(block, res).await
-        };
-
-        let result = inner().await;
+        let res = self.execute_block(block).await?;
```

Comment:
Note that now there will be no log if execution fails. It sounds reasonable but it's not the original behavior.